### PR TITLE
fixbug:when enable api v2 , TiDB resolve lock meet error

### PR DESCRIPTION
**Pull request by ystaticy**
Signed-off-by: ystaticy <y_static_y@sina.com>

### What is changed and how it works?

Issue Number: Close https://github.com/tikv/tikv/issues/12595

What's Changed:

```
If cmd is scan_lock , it will not execute invalid_key_range_mode check.
```
### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
fix:when enable api v2 , TiDB resolve lock meet error
```

### CODE REVIEW

File: src/storage/mod.rs [hunk 1]
```diff
@@ -485,7 +485,9 @@ impl<E: Engine, L: LockManager, F: KvFormat> Storage<E, L, F> {
                         range.0.as_ref().map(AsRef::as_ref),
                         range.1.as_ref().map(AsRef::as_ref),
                     );
-                    if ApiV2::parse_range_mode(range) != KeyMode::TiDB {
+                    if ApiV2::parse_range_mode(range) != KeyMode::TiDB
```
Comment (pingyu) [inline src/storage/mod.rs:488]: Suggest to just remove `check_api_version_ranges` in `scan_lock` [here](https://github.com/tikv/tikv/blob/1fb8980ccab9ff40c1adc206df52952dab8e8ad8/src/storage/mod.rs#L1267). Also leave comments about why.

Comment (ystaticy) [reply src/storage/mod.rs]: OK. I already fix it in ```scan_lock```, and use the old conditional judgment inside ```check_api_version_ranges```.

File: src/storage/mod.rs [hunk 2]
```diff
@@ -1264,15 +1263,8 @@ impl<E: Engine, L: LockManager, F: KvFormat> Storage<E, L, F> {
...
-                    api_version,
-                    ctx.api_version,
-                    CMD,
-                    [(
-                        start_key.as_ref().map(Key::as_encoded),
-                        end_key.as_ref().map(Key::as_encoded),
-                    )],
-                )?;
+                // Remove the check_api_version_ranges,
+                // because of resolve locks range from TiDB is invalid key range mode
```
Comment (pingyu) [inline src/storage/mod.rs:1267]: Suggestion:
Do not `check_api_version` in `scan_lock`, to be compatible with TiDB gc-worker,
which resolves locks on regions, and boundary of regions will be out of range of TiDB keys.

**Review by pingyu — COMMENTED** [review]
Rest LGTM~

**Review by MyonKeminta — APPROVED** [review]

**Review by iosmanthus — APPROVED** [review]
LGTM

File: tests/integrations/storage/test_storage.rs [hunk 3]
```diff
@@ -966,7 +966,7 @@ fn test_txn_store_txnkv_api_version() {
                 store.batch_get_command_err(&[key, key, key], 10);
 
                 store.scan_err(key, None, 100, 10);
-                store.scan_locks_err(20, key, &end_key, 10);
+                store.scan_locks_ok(20, key, &end_key, 10, vec![]);
```
Comment (pingyu) [inline tests/integrations/storage/test_storage.rs:969]: Please comment that why it is ok here.

**Pull request merged**